### DISCUSSION TITLE
关于 maryoko-ui 一些意见征收

### DISCUSSION THREAD

**Discussion by innocces** [maryoku-ui/maryoku-ui#1] [Polls]
<div align="center">
  <img width="150" src="https://cdn.jsdelivr.net/gh/maryoku-ui/maryoku-ui/docs/public/image/favicon.svg" />

  <h2>maryoku-ui</h2>
  <h2><a href="https://discord.gg/N82HK72uJk" target="__blank">discord</a></h2>
</div>

## slogen
以**web**优先的**UI**组件库. 希望它可以给使用者提供源源不断的魔力(máꜜryòkù)

## 背景
为多端组件库增加维护的可能. 主要但不局限于: **h5**、 **Native(RN)** 以及 **wechatmoniprograme(base taro)**.

## 技术栈
考虑到**RN**的特殊性, 并且**Design Token**的便捷性. 样式方面我们选择**tailwind**. (为什么不是**wind**和**Uno**? 没有为啥😜)

**chakra-ui**的使用便捷性要超过**material-ui**(只是我个人觉得).  同样类似**JSS**的原子能力

其实背靠一个现有的并且设计体系较为完备的库. 可以帮我们缩减一定的时间来短时间构建一定规模的组件.  举个例子:

经过定制之后的**chakra**(就是魔化后的**maryoku**). 大部分的组件就可以拥有**mobile**的能力了. 而**RN**部分在一阶段我们可以只提供简单的组件. 高阶部分滞后.

## Q
1. **@emotion/xxx**相关包是否可用于小程序端? (若不能， 需要寻找替代方案)
2. 组件优先级
3. 定制**chakra**
4. 文档选型
5. 代码规范
6. 粒度是否要拆分类似**chakra**到组件层面?
7. **zag**？
8. **chakra.xxx**?
9. 视觉？

## Link
1. [chakra](https://github.com/chakra-ui/chakra-ui)
2. [chakra-doc](https://chakra-ui.com/docs/components)
3. [emotion](https://github.com/emotion-js/emotion)
------
关于你们的想法?

**Comment by innocces** [comment]
关于参与者

如果你有想法一起参与到这个组件库的建设. 请aprove & reply this discussions

**Comment by innocces** [comment]
弃用**NativeUI**的原因是写法在我看来有点反人类

**Comment by innocces** [comment]
要注意的是，**RN**侧还是重**UI**的比较好. 提供组合式的做法让开发者自行利用**UI + hooks**来完成

**Comment by innocces** [comment]
跨端. 目前的想法是分包. 

1. **@maryoku-ui/web**：浏览器
2. **@maryoku-ui/rn**:  RN
3. **@maryoku-ui/taro**: taro ( =  web + rn + mini )

**Reply by akazwz** [reply]
是叫 @maryoku-ui/web 还是 @maryoku-ui/react这种？

**Reply by innocces** [reply]
确实. 会让人觉得**vue**也能用. 对对对

**Reply by ryan-117** [reply]
准备只支持react吗？

**Reply by innocces** [reply]
> 准备只支持react吗？

目前是的

**Comment by TinsFox** [comment]
好像已经有一个chakra 定制的库了https://github.com/horizon-ui/horizon-ui-chakra

**Reply by innocces** [reply]
这个是web端的. maryoku定位是移动端

**Reply by akazwz** [reply]
我认为这个是类似与封装好的dashboard,和这个要搞一个多端的类原子组件还是有区别的

**Reply by innocces** [reply]
> 我认为这个是类似与封装好的dashboard,和这个要搞一个多端的类原子组件还是有区别的

_(:з」∠)_ 对对对

**Reply by TinsFox** [reply]
嗦嘎

**Comment by TinsFox** [comment]
文档选型怎么样啦？

**Reply by innocces** [reply]
> 文档选型怎么样啦？

astro!!!

**Comment by TinsFox** [comment]
不懂就问 chakra-ui的样式哪里来的，@chakra-ui/system吗

**Comment by innocces** [comment]
> 不懂就问 chakra-ui的样式哪里来的，@chakra-ui/system吗

本质上你可以认为 **@maryoku-ui/react** 是 **@chakra-ui/react**的一个超集

**Comment by ryan-117** [comment]
有新进度吗？

**Reply by innocces** [reply]
没有。我在摆烂
